phpunit/interactivity-api/class-wp-interactivity-api-wp-each-test.php

```diff
@@ -0,0 +1,516 @@
+			'<div id="after-wp-each" data-wp-bind--id="myPlugin::state.after">Text</div>';
+		$new      = $this->interactivity->process_directives( $original );
+		$this->assertEquals( $expected, $new );
+	}
+
+	/**
+	 * Tests that the `data-wp-each` directive works with different item names.
+	 *
+	 * @covers ::process_directives
+	 */
+	public function test_wp_each_different_item_names() {
+		$original = '' .
+			'<template data-wp-each--myitem="myPlugin::state.list">' .
+				'<span data-wp-text="myPlugin::context.myitem"></span>' .
+			'</template>' .
+			'<div data-wp-bind--id="myPlugin::state.after">Text</div>';
+		$expected = '' .
+			'<template data-wp-each--myitem="myPlugin::state.list">' .
+				'<span data-wp-text="myPlugin::context.myitem"></span>' .
+			'</template>' .
+			'<span data-wp-each-child data-wp-text="myPlugin::context.myitem">1</span>' .
+			'<span data-wp-each-child data-wp-text="myPlugin::context.myitem">2</span>' .
+			'<div id="after-wp-each" data-wp-bind--id="myPlugin::state.after">Text</div>';
+		$new      = $this->interactivity->process_directives( $original );
+		$this->assertEquals( $expected, $new );
+	}
+
+	/**
+	 * Tests that the `data-wp-each` directive transforms kebab-case into
+	 * camelCase.
+	 *
+	 * @covers ::process_directives
+	 */
+	public function test_wp_each_different_item_names_transforms_camelcase() {
+		$original = '' .
+			'<template data-wp-each--my-item="myPlugin::state.list">' .
+				'<span data-wp-text="myPlugin::context.myItem"></span>' .
+			'</template>' .
+			'<div data-wp-bind--id="myPlugin::state.after">Text</div>';
+		$expected = '' .
+			'<template data-wp-each--my-item="myPlugin::state.list">' .
+				'<span data-wp-text="myPlugin::context.myItem"></span>' .
+			'</template>' .
+			'<span data-wp-each-child data-wp-text="myPlugin::context.myItem">1</span>' .
+			'<span data-wp-each-child data-wp-text="myPlugin::context.myItem">2</span>' .
+			'<div id="after-wp-each" data-wp-bind--id="myPlugin::state.after">Text</div>';
+		$new      = $this->interactivity->process_directives( $original );
+		$this->assertEquals( $expected, $new );
+	}
+
+	/**
+	 * Tests that the `data-wp-each` directive doesn't work with top-level texts.
+	 *
+	 * @covers ::process_directives
+	 */
+	public function test_wp_each_doesnt_work_with_top_level_text() {
+		$original = '' .
+			'<template data-wp-each="myPlugin::state.list">' .
+				'id: <span data-wp-text="myPlugin::context.item"></span>' .
+			'</template>';
+		$new      = $this->interactivity->process_directives( $original );
+		$this->assertEquals( $original, $new );
+
+		$original = '' .
+			'<template data-wp-each="myPlugin::state.list">' .
+				'<span data-wp-text="myPlugin::context.item"></span>!' .
+			'</template>';
+		$new      = $this->interactivity->process_directives( $original );
+		$this->assertEquals( $original, $new );
+	}
+
+	/**
+	 * Tests that the `data-wp-each` directive works with nestded template tags.
```

Comment:
```suggestion
	 * Tests that the `data-wp-each` directive works with nested template tags.
```

Reply:
Thanks! Fixed in https://github.com/WordPress/gutenberg/pull/58498/commits/72a2f19de85622bfa4274aa5074776855c3b38c7.